.github/workflows/gdunit4_tests.yml

```
@@ -1,83 +1,307 @@
+
+          # 3. Clean up batch directory
+          rm -rf lcov_batches
+
+          echo "=== Final Sanitized LCOV Summary ==="
+          lcov --summary final_coverage.lcov
+
+      # Locate most recent XML test report directory generated by GDUnit4
+      - name: "Find Latest Report Directory"
+        id: "find_report"
+        if: "always()"
         run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2012:info:1:17: Use find instead of ls to better handle non-alphanumeric filenames

(shellcheck)